pkg/maps/lbmap/lbmap.go

```diff
@@ -398,6 +399,14 @@ func (*LBBPFMap) UpdateSourceRanges(revNATID uint16, prevSourceRanges []*cidr.CI
 
 	srcRangeMap := map[string]*cidr.CIDR{}
 	for _, cidr := range sourceRanges {
+		// k8s api server does not catch the IP family mismatch, so we need to catch it here
+		if ip.IsIPv6(cidr.IP) == !ipv6 {
+			err := fmt.Errorf("cidr:%v", cidr)
+			log.WithFields(logrus.Fields{
+				logfields.ServiceID: revNATID,
+			}).WithError(err).Warn("Source range's IP family does not match with the LB's")
```

Comment:
```suggestion
			log.WithFields(logrus.Fields{
				logfields.ServiceID: revNATID,
				logfields.CIDR: cidr,
			}).Warn("Source range's IP family does not match with the LB's")
```

I think this will achieve the same visibility without the need to construct an error that is basically acting like a logfield anyway.

Reply:
Aha, make sense. Upated.